utils/src/set.rs

```diff
@@ -1,19 +1,30 @@
 //! Ordered collections that guarantee sorted, deduplicated keys.
 
 #[cfg(not(feature = "std"))]
-use alloc::vec::Vec;
+use alloc::{collections::BTreeSet, vec::Vec};
 use bytes::{Buf, BufMut};
 use commonware_codec::{EncodeSize, RangeCfg, Read, Write};
 use core::{
     fmt,
     ops::{Deref, Index, Range},
 };
+#[cfg(feature = "std")]
+use std::collections::BTreeSet;
+use thiserror::Error;
 
 #[cfg(not(feature = "std"))]
 type VecIntoIter<T> = alloc::vec::IntoIter<T>;
 #[cfg(feature = "std")]
 type VecIntoIter<T> = std::vec::IntoIter<T>;
 
+/// Errors that can occur when constructing [`OrderedAssociatedUnique`].
```

Comment:
Open question: is there a case where we would want `OrderedAssociated` (non-unique)?

In `p2p`, it may be useful to allow multiple people to register the same socket info but that seems odd...